src/ITensorNetworks/peps.jl

```diff
@@ -0,0 +1,142 @@
+"""
+A finite size PEPS type.
+"""
+mutable struct PEPS
+  data::Matrix{ITensor}
+end
+
+PEPS(Nx::Int, Ny::Int) = PEPS(Matrix{ITensor}(undef, Nx, Ny))
+
+"""
+    PEPS([::Type{ElT} = Float64, sites; linkdims=1)
+Construct an PEPS filled with Empty ITensors of type `ElT` from a collection of indices.
+Optionally specify the link dimension with the keyword argument `linkdims`, which by default is 1.
+"""
+function PEPS(::Type{T}, sites::Matrix{<:Index}; linkdims::Integer=1) where {T<:Number}
+  Ny, Nx = size(sites)
+  tensor_grid = Matrix{ITensor}(undef, Ny, Nx)
+  # we assume the PEPS at least has size (2,2). Can generalize if necessary
+  @assert(Nx >= 2 && Ny >= 2)
+
+  lh = Matrix{Index}(undef, Ny, Nx - 1)
+  for ii in 1:(Nx - 1)
+    for jj in 1:(Ny)
+      lh[jj, ii] = Index(linkdims, "Lh,$jj,$ii")
+    end
+  end
+  lv = Matrix{Index}(undef, Ny - 1, Nx)
+  for ii in 1:(Nx)
+    for jj in 1:(Ny - 1)
+      lv[jj, ii] = Index(linkdims, "Lv,$jj,$ii")
+    end
+  end
+
+  # boundary cases
+  tensor_grid[1, 1] = emptyITensor(T, lh[1, 1], lv[1, 1], sites[1, 1])
```

Comment:
Note that you can just use `ITensor` here, `emptyITensor` is not needed any more since `ITensor` will default to `EmptyStorage`.

Reply:
Done